src/filter.rs

```diff
@@ -15,167 +15,237 @@
 // along with this program.  If not, see <http://www.gnu.org/licenses/>.
 use std::collections::HashMap;
 
-use log::info;
-use sqlparser::sqlast::{ASTNode, SQLStatement};
+use sqlparser::sqlast::{ASTNode, SQLBinaryOperator, SQLStatement};
 
 pub fn line_fails_query_conditions(
     line: &String,
     query: &SQLStatement,
-    projection_values: &HashMap<String, String>,
+    projection_values: &HashMap<String, Option<String>>,
 ) -> bool {
     let mut skip_line = false;
     match query {
-        sqlparser::sqlast::SQLStatement::SQLQuery(ref q) => {
-            match q.body {
-                sqlparser::sqlast::SQLSetExpr::Select(ref bodyselect) => {
-                    let mut all_conditions_pass = true;
-                    for slct in &bodyselect.selection {
-                        info!("ordered ops: {:?}", &slct);
-                        match slct {
-                            sqlparser::sqlast::ASTNode::SQLIsNotNull(ast) => {
-                                let identifier = get_identifier_from_ast(&ast);
-                                if projection_values.contains_key(&identifier[..]) == false
-                                    || projection_values[&identifier] == ""
-                                {
-                                    all_conditions_pass = false;
-                                }
+        sqlparser::sqlast::SQLStatement::SQLQuery(ref q) => match q.body {
```

Comment:
Single-arm `match` clauses should be avoided. See the `if let` pattern.

Reply:
the only problem with this specific line is that it's nested inside another match statement, not sure we can match 3 levels deep in a single `if let`

Reply:
Well, we have 3 `match` clauses with only one arm. You can compact that to 3 `if let` which the more natural thing to do in rust. See the PR I sent to your branch: https://github.com/dvaldivia/minsql/pull/4/files

Reply:
Didn't saw that before! I brought that change into this branch since your PR now has conflicts, figured this might be more convenient.